ui/commands.go

```diff
@@ -176,10 +179,82 @@ func cmdRedact(cmd *Command) {
 	cmd.Room.StartSelecting(SelectRedact, strings.Join(cmd.Args, " "))
 }
 
+func autocompleteFile(cmd *CommandAutocomplete) (completions []string, newText string) {
+	inputPath, err := filepath.Abs(cmd.RawArgs)
+	if err != nil {
+		return
+	}
+
+	var searchNamePrefix, searchDir string
+	if strings.HasSuffix(cmd.RawArgs, "/") {
+		searchDir = inputPath
+	} else {
+		searchNamePrefix = filepath.Base(inputPath)
+		searchDir = filepath.Dir(inputPath)
+	}
+	files, err := ioutil.ReadDir(searchDir)
+	if err != nil {
+		return
+	}
+	for _, file := range files {
+		name := file.Name()
+		if !strings.HasPrefix(name, searchNamePrefix) || (name[0] == '.' && searchNamePrefix == "") {
+			continue
+		}
+		fullPath := filepath.Join(searchDir, name)
+		if file.IsDir() {
+			fullPath += "/"
+		}
+		completions = append(completions, fullPath)
+	}
+	if len(completions) == 1 {
+		newText = fmt.Sprintf("/%s %s", cmd.OrigCommand, completions[0])
+	}
+	return
+}
+
 func cmdDownload(cmd *Command) {
 	cmd.Room.StartSelecting(SelectDownload, strings.Join(cmd.Args, " "))
 }
 
+func cmdUpload(cmd *Command) {
+	if len(cmd.Args) == 0 {
+		cmd.Reply("Usage: /upload <file>")
+		return
+	}
+
+	path, err := filepath.Abs(cmd.RawArgs)
+	if err != nil {
+		cmd.Reply("Failed to get absolute path: %v", err)
+		return
+	}
+
+	ftype, err := mimetype.DetectFile(path)
+	if err != nil {
+		cmd.Reply("Failed to get conetnt type: %v", err)
+		return
+	}
+
+	fi, err := os.Stat(path)
+	if err != nil {
+		cmd.Reply("Failed to stat file: %v", err)
+		return
+	}
+
+	reader, err := os.Open(path)
+	if err != nil {
+		cmd.Reply("Failed to open file: %v", err)
+		return
+	}
+
+	cmd.Room.UploadMedia(mautrix.ReqUploadMedia{
```

Comment:
Here I need to get the contentUri and I'm thinking wether to handle it in or return it from https://github.com/tulir/gomuks/pull/206/files#diff-3834a9eb4471606d367d9f087f2b88ddR820

Reply:
How'd you solve it @tulir